pkg/virt-api/api_test.go

```diff
@@ -185,128 +324,415 @@ var _ = Describe("Virt-api", func() {
 		}, 5)
 
 		It("should create apiservice endpoint if one doesn't exist", func(done Done) {
-			app.signingCertBytes = []byte("fake")
-			apiService := &apiregistrationv1beta1.APIService{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: subresourceAggregatedApiName,
-					Labels: map[string]string{
-						v1.AppLabel: "virt-api-aggregator",
-					},
-				},
-				Spec: apiregistrationv1beta1.APIServiceSpec{
-					Service: &apiregistrationv1beta1.ServiceReference{
-						Namespace: namespaceKubevirt,
-						Name:      "virt-api",
-					},
-					Group:                v1.SubresourceGroupName,
-					Version:              v1.SubresourceGroupVersion.Version,
-					CABundle:             app.signingCertBytes,
-					GroupPriorityMinimum: 1000,
-					VersionPriority:      15,
-				},
-			}
+			expectedApiService := app.subresourceApiservice()
+			expectedApiService.Kind = "APIService"
+			expectedApiService.APIVersion = "apiregistration.k8s.io/v1beta1"
 			server.AppendHandlers(
 				ghttp.CombineHandlers(
-					ghttp.VerifyRequest("GET", "/api/v1/namespaces/kubevirt/apiservices/"+subresourceAggregatedApiName),
+					ghttp.VerifyRequest("GET", "/apis/apiregistration.k8s.io/v1beta1/apiservices/"+subresourceAggregatedApiName),
```

Comment:
why did this api path change? Was it wrong to begin with? If so I'm confused how this test would have previously passed.

I wouldn't expect any of the changes in this pr to involve a different api path for getting registered apiservices.

Reply:
The original version of those test were not exercising nothing, they pass because ghttp VerifyRequest are not expectantions, if they don't get executing it does not fail, you can check that code is not exercised [here](https://coveralls.io/builds/20711546/source?filename=pkg%2Fvirt-api%2Fapi.go#L768) 

So in the original version kubecli.GetConfig() fails and that's all, to fix that I have set  the "master" flag to server.URL [here](https://github.com/kubevirt/kubevirt/pull/1853/files/39fbe6816f6ddda0d0dd548808969421cdda5248#diff-146b361f2151a1125ec6d16b7f3c01ecR341)

If the URL does not make sense, maybe I set the "master" wrongly, but code is being exerced now, so it has to be good.

Reply:
ha, good catch :+1: